pwem/viewers/viewer_localres.py

```diff
@@ -108,3 +117,213 @@ def createChimeraScript(self, scriptFile, fnResVol,
+                          'parameters below. If the highest and lowest values you input are the same, '
+                          'the lowest and highest values in the date are used.'
+                     )
+
+      from pwem.wizards.wizard import ColorScaleWizardBase
+      group = form.addGroup('Color settings')
+      ColorScaleWizardBase.defineColorScaleParams(group, defaultLowest=0, defaultHighest=2, defaultIntervals=21,
+                                                  defaultColorMap='RdBu')
+
+    def _getOutputObject(self):
+        '''Return the output object of the protocol (SetOfAtomStruct or AtomStruct)
+        To change in other viewers with different output name'''
+        return getattr(self.protocol, self.protocol._OUTNAME)
+
+    def getAtomStructObject(self):
+        obj = self._getOutputObject()
+        if issubclass(type(obj), SetOfAtomStructs):
+          _inputStruct = obj.getRepresentative()
+          if _inputStruct is None:
+              _inputStruct = obj.getFirstItem()
+        elif issubclass(type(obj), AtomStruct):
+          _inputStruct = obj
+        return _inputStruct
+
+    def _getStructureAttributes(self):
+        '''Returns a list with the names of the attributes of the output object'''
+        obj = self.getAtomStructObject()
+        ASH = AtomicStructHandler()
+        cifDic = ASH.readLowLevel(obj.getFileName())
+        return list(set(cifDic[NAME]))
+
+    def _getAttributeIndex(self, attrName):
+        '''Returns a list with the names of the attributes of the output object'''
+        names = self._getStructureAttributes()
+        return names.index(attrName)
+
+
+    def _getVisualizeDict(self):
+      return {
+        'displayStruct': self._showChimera,
+        'displayHistogram': self._showHistogram,
+      }
+
+    def _showChimera(self, paramName=None):
+      prot = self.protocol
+      chimScript = prot._getExtraPath('chimera_script.py')
+      f = open(chimScript, "w")
+      f.write("from chimerax.core.commands import run\n")
+
+      f.write("run(session, 'cd %s')\n" % os.getcwd())
+      f.write("run(session, 'cofr 0,0,0')\n")  # set center of coordinates
+
+      _inputStruct = self.getAtomStructObject()
+      _inputVol = _inputStruct.getVolume()
+      if _inputVol is not None:
+        strId = 3
+        dim, sampling = _inputVol.getDim()[0], _inputVol.getSamplingRate()
+
+        f.write("run(session, 'open %s')\n" % _inputVol.getFileName())
+        x, y, z = _inputVol.getOrigin(force=True).getShifts()
+        f.write("run(session, 'volume #%d style surface voxelSize %f')\n" % (1, sampling))
+        f.write("run(session, 'volume #%d origin %0.2f,%0.2f,%0.2f')\n" % (1, x, y, z))
+        f.write("run(session, 'hide #!%d models')\n" % 1)
+
+      else:
+        dim, sampling = 150., 1.
+        strId = 2
+
+      tmpFileName = os.path.abspath(prot._getExtraPath("axis_input.bild"))
+      Chimera.createCoordinateAxisFile(dim,
```

Comment:
What if Chimera plugin is missing?

Reply:
this problem was present already as the LocalResolutionViewer also uses some Chimera funstions.
Should we make more robust the import for both of them (and maybe launch a warning window for both of the viewers if chimera is missing?)